flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterGateway.java

```diff
@@ -218,13 +220,27 @@ CompletableFuture<String> triggerSavepoint(
             final SavepointFormatType formatType,
             @RpcTimeout final Time timeout);
 
+    /**
+     * Triggers taking a checkpoint of the executed job.
+     *
+     * @param checkpointProperties to determine how checkpoint should be taken or null if the
+     *     existing checkpoint property should be used
+     * @param timeout for the rpc call
+     * @return Future which is completed with the savepoint path once completed
```

Comment:
nit: java doc should say it returns `{@link CompletedCheckpoint}` not `path`.

Reply:
Good catch